push_new_pricing_to_statusdb.py

```diff
@@ -0,0 +1,288 @@
+
+# The combination of these "columns" need to be unique within the document
+UNIQUE_KEY_SETS = {'products': ('ID', ['Category', 'Type', 'Name']),
+                   'components': ('REF_ID', ['Category', 'Type', 'Product name', 'Units'])}
+
+MAX_NR_ROWS = 200
+
+# Assuming the rows of products are sorted in the preferred order
+
+# Set up a logger with colored output
+logger = logging.getLogger(__name__)
+logger.propagate = False  # Otherwise the messages appeared twice
+coloredlogs.install(level='INFO', logger=logger,
+                    fmt='%(asctime)s %(levelname)s %(message)s')
+
+
+def check_unique(items, type):
+    """Make sure all items within _items_
+
+    fulfill the uniqueness criteria according to the UNIQUE_KEY_SETS
+    """
+    key_val_set = set()
+    for id, item in items.items():
+        id_key, keys = UNIQUE_KEY_SETS[type]
+        t = tuple(item[key] for key in keys)
+
+        # Check that it is not already added
+        if t in key_val_set:
+            raise ValueError("Key combination {}:{} is included multiple "
+                             "times in the {} sheet. "
+                             "ABORTING.".format(keys, t, type))
+        key_val_set.add(t)
+    return True
+
+
+def check_conserved(new_items, current_items, type):
+    """Ensures the keys in CONSERVED_KEY_SETS are conserved for each given id.
+
+    Compares the new version against the currently active one.
+    Params:
+        new_items     - A dict of the items that are to be added
+                        with ID attribute as the key.
+        current_items - A dict of the items currently in the database
+                        with ID attribute as the key.
+        type          - Either "components" or "products"
+    """
+    conserved_keys = CONSERVED_KEY_SETS[type][1]
+
+    for id, new_item in new_items.items():
+        if str(id) in current_items:
+            for conserved_key in conserved_keys:
+                if new_item[conserved_key] != current_items[str(id)][conserved_key]:
+                    raise ValueError("{} need to be conserved for {}."
+                                     " Violated for component with id {}. "
+                                     "Found {} for new and {} for current. "
+                                     "ABORTING!".format(
+                                        conserved_key,
+                                        type,
+                                        id,
+                                        new_item[conserved_key],
+                                        current_items[str(id)][conserved_key]
+                                        ))
+
+
+def get_current_items(db, type):
+    rows = db.view("entire_document/by_version", descending=True, limit=1).rows
+    if len(rows) != 0:
+        doc = rows[0].value
+        return doc[type]
+    return {}
+
+
+def is_empty_row(comp):
+    for k, v in comp.items():
+        if v != '':
```

Comment:
Can `v` be `None` or any other object than `str` ? and if it can then it would be considered as not empty (?)